science/gnss-sdr/files/patch-boost-asio.diff

```diff
@@ -0,0 +1,11 @@
+--- CMakeLists.txt.orig
++++ CMakeLists.txt
+@@ -503,7 +503,7 @@
+     message(FATAL_ERROR "Fatal error: Boost (version >=${GNSSSDR_BOOST_MIN_VERSION}) required.")
+ endif()
+ 
+-
++add_definitions(-DBOOST_ASIO_HAS_STD_STRING_VIEW=1)
```

Comment:
I think this change has other conditions. In a forthcoming patch for UHD, I use the following (I need to test 1.65.0 to see if it requires this too):
```
# for Boost 1.6[6-8].0, disable std::string_view entirely
# if the -std=c++14 or earlier.
if((${Boost_VERSION} VERSION_EQUAL 106600) OR
   (${Boost_VERSION} VERSION_EQUAL 106700) OR
   (${Boost_VERSION} VERSION_EQUAL 106800))
  if(${CMAKE_CXX_STANDARD} LESS 17)
    add_definitions(-DBOOST_ASIO_HAS_STD_STRING_VIEW=0)
  endif()
endif()
```
I'm not sure if gnss-sdr requires something similar, not just a blanket statement.

Reply:
I don't know exacly why, but this fails in my machine when CMAKE_CXX_STANDARD is set to 14. I'm trying this other patch, which seems save enough:

``` 
if(${Boost_VERSION} VERSION_LESS 107000)
    # Check if we have std::string_view
    include(CheckCXXSourceCompiles)
    check_cxx_source_compiles("
        #include <string_view>
        int main()
        { std::string_view sv; }"
        has_string_view
    )
    if(${has_string_view})
        add_definitions(-DBOOST_ASIO_HAS_STD_STRING_VIEW=1)
    else()
        add_definitions(-DBOOST_ASIO_HAS_STD_STRING_VIEW=0)
    endif()
endif()
```

This seems fixed in Boost 1.70: https://github.com/boostorg/asio/commit/43874d5497414c67655d901e48c939ef01337edb